### DISCUSSION TITLE
SDXL Turbo

### DISCUSSION THREAD

**Discussion by ut1kaa** [lllyasviel/Fooocus#1060] [Ideas]
It would be great to see in Fooocus this model that generates an image in one step, +- in real time.

Model: https://huggingface.co/stabilityai/sdxl-turbo

**Comment by xhoxye** [comment]
I've seen people using it in fooocus

**Reply by Green-li** [reply]
How to use it in Fooocus? 
Any config or code changes?

**Reply by thiner** [reply]
I haven't teied. But from SD's description, just download the weight file, and put it into checkpoints folder should work.

**Reply by AkoZ** [reply]
i've put the adversial diffusion distillation .safetensors ( https://huggingface.co/stabilityai/sdxl-turbo ) seems not enough to play with .. the meaning thing is about sigmas .. with a "turbo scheduler": seen there : https://www.youtube.com/watch?v=DZ2dfq8ljrc
oops EDIT .. not seen the next com"' ! :)

**Comment by joanorsky** [comment]
I am also curious about this...

**Comment by xhoxye** [comment]
> How to use it in Fooocus?如何在Fooocus中使用它？ Any config or code changes?任何配置或代码更改？
However, he uses a Chinese interface to operate.
![QQ截图20231129180402](https://github.com/lllyasviel/Fooocus/assets/129571231/dcbef333-dcf9-4133-a769-dee21c1a39dd)
![QQ截图20231129180305](https://github.com/lllyasviel/Fooocus/assets/129571231/7b57ab8e-ae8a-4d8c-80dd-8f064a9ca077)
![QQ截图20231129180245](https://github.com/lllyasviel/Fooocus/assets/129571231/1c675291-9612-40ef-905f-1972926d7a04)

**Comment by ut1kaa** [comment]
I would like to see that the picture when selecting this model updated in real time or something like that, after the addition of the promt, that is, we change the promt and the picture is updated, we add something and the picture on the same seed is updated and the token we need is added. And with a separate button, we would randomize the seed for this to get something different. I hope I've made myself clear.

**Comment by jepjoo** [comment]
I highly recommend TurboVisionXL finetune. **1024x1024** works well, LoRAs and Controlnets work too. 

**use the settings recommended on the model page**:
https://civitai.com/models/215418?modelVersionId=242733

![Screenshot 2023-11-29 at 14-31-28 Fooocus 2 1 824 oma](https://github.com/lllyasviel/Fooocus/assets/48107422/264f8d97-774d-4146-a349-9cb4fa562704)

**Comment by HynoR** [comment]
Here is the short guide for using SDXL-turbo model in the Fooocus.
Download Turbo Model and drop it into the checkpoints folder -> https://huggingface.co/stabilityai/sdxl-turbo

In web panel:
1. Select Model Option, Change Base Model to SDXL Turbo, and Change Refiner to None. 
![image](https://github.com/lllyasviel/Fooocus/assets/20227709/a884dd23-a732-4fd7-9291-7a6f541c36aa)

2. Select Advanced Option, Edit Guidance Scale to 1. Forced Overwrite of Sampling Step to 1. Forced Overwrite of Generating Width/Height to 512.
<img width="417" alt="image" src="https://github.com/lllyasviel/Fooocus/assets/20227709/765f0e69-5e10-4259-ae9b-60c016ac09cc">

![image](https://github.com/lllyasviel/Fooocus/assets/20227709/3a708b42-b426-46a4-834e-60c98e6cc6eb)

4. Input your prompt and enjoy! Meow!

<img width="863" alt="image" src="https://github.com/lllyasviel/Fooocus/assets/20227709/29715677-586a-4d45-a3e9-8cd538d66f63">

**Reply by thiner** [reply]
Awesome! I tried the model, seems only one step is not enough for drawing human, the hands and face problems are quite obviouse.

**Reply by HynoR** [reply]
![image](https://github.com/lllyasviel/Fooocus/assets/20227709/e7e4b25b-b8b0-4d60-ac36-02d31ca1b032)
From SDXL_Turbo Readme. It is normal that it cannot be generated human face correctly.

**Reply by deadpipe** [reply]
Thankyou @HynoR , those settings worked perfectly

**Reply by Uneternal** [reply]
I'm putting exactly these settings and Fooocus always just outputs garbage as soon as I put a prompt with weights like (long hair:1.5). Is that normal?

**Comment by thiner** [comment]
I tried the Turbo model for a while. The original output with Fooocus default setting seems not good for the prompts I tried. 
Then I tried the `lcm` samplers, the output is Wow!

**CFG Scale: 1
Size: 512*512, 
Steps: 2**
**Style: "Fooocus V2"**
**Sampler: lcm
Scheduler: lcm**
Prompt copied from: https://civitai.com/images/3963914
![lcm_sampler_2steps_2](https://github.com/lllyasviel/Fooocus/assets/1897227/93ef99b7-e043-4efa-b427-24c24359ba41)

This is amazing with just two steps, and the original model!

**Reply by joanorsky** [reply]
I guess the turbo is actually useless as it is because we have LCM with much higher quality. As seen above the step count difference is minimal...

**Comment by yesbroc** [comment]
a realtime generation feature would be nice, especially for testing prompts and such or on cpu-only devices.

**Comment by ItsNoted** [comment]
I was able to achieve these results using the DreamShaper XL Turbo SDXL checkpoint.

https://prompts.mindrenders.com/m/22

**Reply by jepjoo** [reply]
I tried DreamShaper XL Turbo too and it seems a somewhat better with eyes than other Turbo merges. I'd say it's the best Turbo model so far but haven't really tested comprehensively. Link for those interested:

https://civitai.com/models/112902?modelVersionId=251662

**Comment by VictorZakharov** [comment]
I am getting crappy results like this:
![image](https://github.com/lllyasviel/Fooocus/assets/8978766/23d107b0-f751-49f4-ba99-794dd1fa3587)
![image](https://github.com/lllyasviel/Fooocus/assets/8978766/87d9ece5-f59b-454b-9aa9-e907b38a3659)

I tried karras, turbo for scheduler,
sampler turbo or lcm - no difference.
model is turbo vision XL.

**Reply by joanorsky** [reply]
Guidance scale too high.. set it from 1 to 2.25
Make it 3 to 5 steps for ok or 6 to 10 for better results even
Sampler make it  DPM++ SDE or DPM++ SDE Karras

Better coherence on 1024x1024

**Reply by VictorZakharov** [reply]
@joanorsky CFG: 1
sampler: dpmpp_2m_sde_gpu
scheduler: turbo
steps: 4
1024x1024

This is not good, still.
![image](https://github.com/lllyasviel/Fooocus/assets/8978766/08041094-e021-4e33-b8e2-eb7bf9797acf)

**Reply by VictorZakharov** [reply]
10 steps, same settings:
![image](https://github.com/lllyasviel/Fooocus/assets/8978766/05391c8a-044b-4a25-8092-6594b43faf47)

**Reply by VictorZakharov** [reply]
On CFG 2 it has less colorful noise (or none), but more distortion in the image -- the woman likely to become ugly.

**Reply by VictorZakharov** [reply]
Ok, so these props seem to give decent results. And now 10 steps actually improves it, rather than adding noise.
![image](https://github.com/lllyasviel/Fooocus/assets/8978766/e628a6ee-5a44-4bfb-83f6-14c375409872)
![image](https://github.com/lllyasviel/Fooocus/assets/8978766/de8cb9e0-2623-4f12-853c-1d07e1d48793)
![image](https://github.com/lllyasviel/Fooocus/assets/8978766/6a7af26d-d6ce-41e0-8c23-1ca7e5cc2aef)
![image](https://github.com/lllyasviel/Fooocus/assets/8978766/57cd91f6-4bb9-41d8-b89c-30ab8ac51bcf)

The results:
![image](https://github.com/lllyasviel/Fooocus/assets/8978766/d8a0683b-c3f5-4fcf-9f46-c486f2932144)

**Reply by joanorsky** [reply]
more steps.. better quality... up to 10 steps.. ;)

**Reply by VictorZakharov** [reply]
@joanorsky Yes, once other settings are correct, 10 steps gives about the same quality as 60 steps regular SDXL. Example below. Thanks!
![image](https://github.com/lllyasviel/Fooocus/assets/8978766/a4555c8d-b8a7-4440-b0e4-357f2dbe5a25)

**Reply by joanorsky** [reply]
Looking good! :)
